app/models/crop.rb

```diff
@@ -152,27 +152,15 @@ def default_photo
   # key: sunniness (eg. 'sun')
   # value: count of how many times it's been used by plantings
   def sunniness
-    sunniness = Hash.new(0)
-    plantings.each do |p|
-      if !p.sunniness.blank?
-        sunniness[p.sunniness] += 1
-      end
-    end
-    return sunniness
+    count_uses_of_property 'sunniness'
   end
 
   # crop.planted_from
   # returns a hash of propagation methods (seed, seedling, etc),
   # key: propagation method (eg. 'seed')
   # value: count of how many times it's been used by plantings
   def planted_from
```

Comment:
Where are these count methods called from?

They may be better suited to a ```SELECT COUNT(0), planted_from FROM plantings WHERE crop_id =?``` or similar